src/types/typeGuards/compound.ts

```diff
@@ -1,7 +1,116 @@
 import * as ts from "typescript";
 
+import { isTypeFlagSet } from "../../flags.js";
 import { isTupleType, isTypeReference } from "./objects.js";
 
+/**
+ * `LiteralType` from typescript except that it allows for it to work on arbitrary types.
+ *
+ * @category Type Types
+ */
+export interface UnknownLiteralType extends ts.Type {
```

Comment:
The naming here confused me at first a bit because _"unknown"_ normally refers to the `unknown` keyword. Is there something else we can call it?

...I can't think of anything better, and I guess it makes sense as-is. 🤷

Reply:
I has it as "general" at first but wasn't like that name.
I decided on "unknown" because `value`'s type is `unknown`.